tests/unit/acquisition/multi_objective/test_partition.py

```diff
@@ -23,6 +23,7 @@
 from trieste.acquisition.multi_objective.partition import (
     DividedAndConquerNonDominated,
     ExactPartition2dNonDominated,
+    HypervolumeBoxDecompositionIncrementalDominated,
```

Comment:
Stupid question: how will this actually be used in a real optimization problem? Our multi-objective code currently all calls prepare_default_non_dominated_partition_bounds.

Reply:
sorry should explain more clearly,  it is the first step towards modifying `prepare_default_non_dominated_partition_bounds`, 

In an [upcoming PR](https://github.com/TsingQAQ/trieste/blob/1895f8d5de136fe6758b076bdde222ce9f47636c/trieste/acquisition/multi_objective/partition.py#L95-L96) (which implement an additional functionality (i.e., `FlipTrickPartitionNonDominated`) on top of `HypervolumeBoxDecompositionIncrementalDominated`), we will replace the current used [`DividedAndConquerNonDominated`](https://github.com/secondmind-labs/trieste/blob/3d16566218c8bc78ea86d37953311bce8746c0fe/trieste/acquisition/multi_objective/partition.py#L95-L96) by [`FlipTrickPartitionNonDominated`](https://github.com/TsingQAQ/trieste/blob/1895f8d5de136fe6758b076bdde222ce9f47636c/trieste/acquisition/multi_objective/partition.py#L95-L96), which is expected to provide a faster non-dominated space partition for num of objectives >=3 (hence faster running of code).

These two PRs are separate in the hope to minimize a chunk of code that needs to be reviewed at once, if it is easier to review as a whole to clear the usage of the new code, we may also directly start from the [upcoming PR](https://github.com/TsingQAQ/trieste/blob/1895f8d5de136fe6758b076bdde222ce9f47636c/trieste/acquisition/multi_objective/partition.py#L95-L96).